src/message.rs

```diff
@@ -61,3 +94,75 @@ impl SignedMessage {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use fvm_shared::address::set_current_network;
+
+    #[test]
+    fn test_parse_mainnet_address() {
+        let addr_str = "f1alg2sxw32ns3ech2w7r3dmp2gl2fputkl7x7jta";
+        let addr = parse_address(addr_str, Network::Mainnet).unwrap();
+
+        assert_eq!(addr.to_string(), addr_str);
+    }
+
+    #[test]
+    fn test_parse_testnet_address() {
+        let addr_str = "t410f2oekwcmo2pueydmaq53eic2i62crtbeyuzx2gmy";
+        let addr = parse_address(addr_str, Network::Testnet).unwrap();
+
+        set_current_network(Network::Testnet); // Required to correctly stringify address
+        assert_eq!(addr.to_string(), addr_str);
+    }
+
+    #[test]
+    fn test_parse_wrong_network() {
+        let m_addr_str = "f1alg2sxw32ns3ech2w7r3dmp2gl2fputkl7x7jta";
+        let err = parse_address(m_addr_str, Network::Testnet).unwrap_err();
+        assert_eq!(err.to_string(), "Wrong Network");
+
+        let t_addr_str = "t410f2oekwcmo2pueydmaq53eic2i62crtbeyuzx2gmy";
+        let err = parse_address(t_addr_str, Network::Mainnet).unwrap_err();
+        assert_eq!(err.to_string(), "Wrong Network");
+    }
+
+    #[test]
+    fn test_parse_eth_address_testnet() {
+        let addr_str = "0xd388ab098ed3e84c0d808776440b48f685198498";
+        let addr = parse_address(addr_str, Network::Testnet).unwrap();
+
+        let exp_addr_str = "t410f2oekwcmo2pueydmaq53eic2i62crtbeyuzx2gmy";
+        let exp_addr = parse_address(exp_addr_str, Network::Testnet).unwrap();
+
+        assert_eq!(exp_addr, addr);
+    }
+
+    #[test]
+    fn test_parse_eth_address_mainnet() {
+        let addr_str = "0xd388ab098ed3e84c0d808776440b48f685198498";
+        let addr = parse_address(addr_str, Network::Mainnet).unwrap();
+
+        let exp_addr_str = "f410f2oekwcmo2pueydmaq53eic2i62crtbeyuzx2gmy";
+        let exp_addr = parse_address(exp_addr_str, Network::Mainnet).unwrap();
+
+        assert_eq!(exp_addr, addr);
+    }
+
+    #[test]
+    fn test_parse_eth_address_too_short() {
+        let addr_str = "0xd3";
+        let e = parse_address(addr_str, Network::Mainnet).err().unwrap();
+
+        assert_eq!(e.to_string(), "Invalid Address");
+    }
```

Comment:
nit: would be great to cover all paths in `check_address_prefix` as well, i.e., test on empty and/or single char input.

Reply:
Your wish is my command 🙇